lib/main_common.pm

```diff
@@ -212,6 +254,143 @@ sub replace_opensuse_repos_tests {
     loadtest "console/zypper_ar";
 }
 
+sub is_memtest {
+    return get_var('MEMTEST');
+}
+
+sub is_mediacheck {
+    return get_var('MEDIACHECK');
+}
+
+sub is_server {
+    return 1 if is_sles4sap();
+    return 1 if get_var('FLAVOR', '') =~ /^Server/;
+    return 0 unless is_leanos();
+    return check_var('SLE_PRODUCT', 'sles');
+}
+
+sub is_desktop {
+    return get_var('FLAVOR', '') =~ /^Desktop/ || check_var('SLE_PRODUCT', 'sled');
+}
+
+sub is_leanos {
+    return 1 if get_var('FLAVOR', '') =~ /^Leanos/;
+    return 1 if get_var('FLAVOR', '') =~ /^Installer-/;
+    return 0;
+}
+
+sub is_desktop_module_selected {
+    # desktop applications module is selected if following variables have following values:
+    # productivity and ha require desktop applications, so it's preselected
+    # same is true for sles4sap
+    return
+         get_var('ADDONS', '') =~ /all-packages|desktop|we/
+      || get_var('WORKAROUND_MODULES', '') =~ /desktop|we/
+      || get_var('ADDONURL',           '') =~ /desktop|we/
+      || get_var('SCC_ADDONS',         '') =~ /desktop|we|productivity|ha/
+      || is_sles4sap;
+}
+
+sub default_desktop {
+    return undef   if get_var('VERSION', '') lt '12';
+    return 'gnome' if get_var('VERSION', '') lt '15';
+    # with SLE 15 LeanOS only the default is textmode
+    return 'gnome' if get_var('BASE_VERSION', '') =~ /^12/;
+    return 'textmode' if (get_var('SYSTEM_ROLE') && !check_var('SYSTEM_ROLE', 'default'));
+    return 'gnome' if is_desktop_module_selected;
```

Comment:
most probably it works as expected after all but just FYI that is_sles4sap defined as positive condition in both is_desktop_module_selected and is_server. So switching lines 302 and 300 will bring different results for SAP guys

Reply:
do you think me just moving it to a different file should have an effect like this?

Reply:
don't things , my comment regarding version_utils not related to this. actually after discussing with you in person I can say that you can ignore this comment because it is just code which you move from one place to another.